app/controllers/cart_controller.rb

```diff
@@ -3,7 +3,12 @@ class CartController < ApplicationController
 
   def add_item
     item = Item.find(params[:item_id])
-    add_item_process(item)
+    if !item.active?
+      flash[:notice] = "This item is inactive"
+      redirect_to "/items/#{item.id}"
+    else
+      add_item_process(item)
+    end
```

Comment:
Add conditional so a flash message will be show and the item will not be added to the cart if the item is inactive